code/modules/mob/living/basic/slime/defense.dm

```diff
@@ -0,0 +1,95 @@
+
+/mob/living/basic/slime/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_SELF)
+		return
+	powerlevel = 0 // oh no, the power!
+
+///If a slime is attack with an empty hand, shoves included, try to wrestle them off the mob they are on
+/mob/living/basic/slime/proc/on_attack_hand(mob/living/basic/slime/defender_slime, mob/living/attacker)
+	SIGNAL_HANDLER
+
+	if(isnull(buckled))
+		return
+
+	if(buckled == attacker ? prob(60) : prob(30)) //its easier to remove the slime from yourself
+		attacker.visible_message(span_warning("[attacker] attempts to wrestle \the [defender_slime.name] off [buckled == attacker ? "" : buckled] !"), \
+		span_danger("[buckled == attacker ? "You attempt" : "[attacker] attempts" ] to wrestle \the [defender_slime.name] off [buckled == attacker ? "" : buckled]!"))
+		playsound(loc, 'sound/weapons/punchmiss.ogg', 25, TRUE, -1)
+		return
+
+	attacker.visible_message(span_warning("[attacker] manages to wrestle \the [defender_slime.name] off!"), span_notice("You manage to wrestle \the [defender_slime.name] off!"))
+	playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
+
+	defender_slime.discipline_slime()
+
+/mob/living/basic/slime/attackby(obj/item/attacking_item, mob/living/user, params)
```

Comment:
i think you should split this up into individual procs that we invoke based on the results of each conditional

Reply:
I have broken this into many, many procs.